### PR TITLE
fix: allow empty IP when decoding Ping's from field

### PR DESCRIPTION
**Pull request by Rjected**
This sets the IP to `UNSPECIFIED` when `0x80` is encountered as a `Ping` packet `from` field.

### CODE REVIEW

File: crates/net/discv4/Cargo.toml [hunk 1]
```diff
@@ -31,7 +31,7 @@ secp256k1 = { workspace = true, features = [
 enr.workspace = true
 
 # async/futures
-tokio = { workspace = true, features = ["io-util", "net", "time"] }
+tokio = { workspace = true, features = ["io-util", "net", "time", "rt-multi-thread"] }
```
Comment (mattsse) [inline crates/net/discv4/Cargo.toml:34]: do we need this?

File: crates/net/discv4/src/proto.rs [hunk 2]
```diff
@@ -191,6 +194,72 @@ pub struct Packet {
     pub hash: B256,
 }
 
+/// Represents the `from` field in the `Ping` packet
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, RlpEncodableWrapper)]
+pub struct PingNodeEndpoint(pub NodeEndpoint);
```
Comment (mattsse) [inline crates/net/discv4/src/proto.rs:199]: does this have to be pub? we only use this internally?

```suggestion
struct PingNodeEndpoint(NodeEndpoint);
```

File: crates/net/discv4/src/proto.rs [hunk 3]
```diff
@@ -191,6 +194,72 @@ pub struct Packet {
...
+        Self(value)
+    }
+}
+
+impl PingNodeEndpoint {
+    /// Creates a new [`PingNodeEndpoint`] from a given UDP address and TCP port.
+    pub const fn from_udp_address(udp_address: &std::net::SocketAddr, tcp_port: u16) -> Self {
+        Self(NodeEndpoint { address: udp_address.ip(), udp_port: udp_address.port(), tcp_port })
+    }
+}
```
Comment (mattsse) [inline crates/net/discv4/src/proto.rs:261]: do we need these?

we only need the `.0`?

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/net/discv4/src/proto.rs (hunk 3 above)
Comment (Rjected) [reply crates/net/discv4/src/proto.rs:261]: don't need these

File: crates/net/discv4/Cargo.toml (hunk 1 above)
Comment (Rjected) [reply crates/net/discv4/Cargo.toml:34]: probably not, left over from something being messed up with my setup, for some reason I can't test discv4 without it

**Review by mattsse — APPROVED** [review]

**Pull request merged**